src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpApplicationController.cs

```diff
@@ -103,46 +103,62 @@ public async Task<IActionResult> Applications()
             var applications = await _apiClient.GetApplications(signinId, false);
             applications = applications.Where(app => app.ApplicationStatus != ApplicationStatus.Rejected).ToList();
 
-            if (!applications.Any())
+            var application = new Apply();
+            Guid applicationId;
+            string applicationStatus;
+
+            if (applications.Any())
             {
-                return await StartApplication(signinId);
+                if (applications.Count > 1)  return View(applications);
+                application = applications.Single();
+                applicationId = application.Id;
+                applicationStatus = application.ApplicationStatus;
             }
+            else
+            {
+                applicationId = await StartApplication(signinId);
+                applicationStatus = ApplicationStatus.InProgress;
 
-            if (applications.Count > 1)
-                return View(applications);
-
-            var application = applications.First();
+                if (applicationId == Guid.Empty)
+                {
+                    return RedirectToAction("EnterApplicationUkprn", "RoatpApplicationPreamble");
+                }
+            }
+            
+            _logger.LogInformation("Applications controller action completed");
```

Comment:
What's the benefit of log entry as it doesn't make much sense to me? And could we downgrade it to Debug?

Reply:
Made them debug